public/app/features/alerting/unified/components/rules/RuleDetailsMatchingInstances.test.tsx

```diff
@@ -0,0 +1,128 @@
+      expect(ui.grafanaStateButton.pending.get(stateFilter)).toBeInTheDocument();
+      expect(ui.grafanaStateButton.noData.get(stateFilter)).toBeInTheDocument();
+      expect(ui.grafanaStateButton.error.get(stateFilter)).toBeInTheDocument();
+    });
+
+    it.each(Object.values(GrafanaAlertState))('Should filter grafana rules by %s state', async (state) => {
+      const rule = mockCombinedRule({
+        promRule: mockPromAlertingRule({
+          alerts: [
+            mockPromAlert({ state: GrafanaAlertState.Normal }),
+            mockPromAlert({ state: GrafanaAlertState.Alerting }),
+            mockPromAlert({ state: GrafanaAlertState.Pending }),
+            mockPromAlert({ state: GrafanaAlertState.NoData }),
+            mockPromAlert({ state: GrafanaAlertState.Error }),
+          ],
+        }),
+      });
+
+      const buttons = {
+        [GrafanaAlertState.Normal]: ui.grafanaStateButton.normal,
+        [GrafanaAlertState.Alerting]: ui.grafanaStateButton.alerting,
+        [GrafanaAlertState.Pending]: ui.grafanaStateButton.pending,
+        [GrafanaAlertState.NoData]: ui.grafanaStateButton.noData,
+        [GrafanaAlertState.Error]: ui.grafanaStateButton.error,
+      };
+
+      render(<RuleDetailsMatchingInstances rule={rule} />);
+
+      await userEvent.click(buttons[state].get());
+
+      expect(ui.instanceRow.getAll()).toHaveLength(1);
+      expect(ui.instanceRow.get()).toHaveTextContent(alertStateToReadable(state));
+    });
+
+    it('For Cloud rules instances filter should contain two states', () => {
+      const rule = mockCombinedRule({
+        namespace: mockPromNamespace(),
+      });
+
+      render(<RuleDetailsMatchingInstances rule={rule} />);
+
+      const stateFilter = ui.stateFilter.get();
+      expect(stateFilter).toBeInTheDocument();
+
+      const stateButtons = ui.stateButton.getAll(stateFilter);
+
+      expect(stateButtons).toHaveLength(2);
+
+      expect(ui.cloudStateButton.firing.get(stateFilter)).toBeInTheDocument();
+      expect(ui.cloudStateButton.pending.get(stateFilter)).toBeInTheDocument();
+    });
+
+    it.each([PromAlertingRuleState.Pending, PromAlertingRuleState.Firing] as const)(
+      'Should filter cloud rules by %s state',
+      async (state) => {
+        const rule = mockCombinedRule({
+          namespace: mockPromNamespace(),
+          promRule: mockPromAlertingRule({
+            alerts: [
+              mockPromAlert({ state: PromAlertingRuleState.Firing }),
+              mockPromAlert({ state: PromAlertingRuleState.Pending }),
+            ],
+          }),
+        });
+
+        render(<RuleDetailsMatchingInstances rule={rule} />);
+
+        await userEvent.click(ui.cloudStateButton[state].get());
+
+        expect(ui.instanceRow.getAll()).toHaveLength(1);
+        expect(ui.instanceRow.get()).toHaveTextContent(alertStateToReadable(state));
+      }
+    );
+  });
+});
+
+function mockPromNamespace(): CombinedRuleNamespace {
```

Comment:
can we use https://github.com/grafana/grafana/blob/5d65d4ce03f18b99891d6fa3e8e58267d4d8ac15/public/app/features/alerting/unified/mocks.ts#L172 instead?